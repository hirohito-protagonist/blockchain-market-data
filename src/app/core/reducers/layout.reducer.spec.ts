import { fromLayoutActions } from './../actions/index.action';
import { reducer, LayoutView } from './layout.reducer';

describe('Core layout store state', () => {

  describe('undefined action', () => {

    it('should return default state', () => {

      const action = { type: void(0) } as fromLayoutActions.ActionType;

      const result = reducer(void(0), action);

      expect(result).toMatchSnapshot();
    });
  });

  describe('[Core Layout] Change View', () => {

    it('should change active view state', () => {

      const state1 = reducer(void(0), new fromLayoutActions.ChangeView(LayoutView.Statistic));

      expect(state1).toMatchSnapshot();

      const state2 = reducer(state1, new fromLayoutActions.ChangeView(LayoutView.ExchangeRate));

      expect(state2).toMatchSnapshot();
    });
  });
});
