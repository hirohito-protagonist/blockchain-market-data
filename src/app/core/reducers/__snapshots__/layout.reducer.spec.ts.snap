// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Core layout store state [Core Layout] Change View should change active view state 1`] = `
Object {
  "activeView": "Statistic",
}
`;

exports[`Core layout store state [Core Layout] Change View should change active view state 2`] = `
Object {
  "activeView": "Exchange rate",
}
`;

exports[`Core layout store state undefined action should return default state 1`] = `
Object {
  "activeView": "Exchange rate",
}
`;
